pkg/gameservers/migration.go

```diff
@@ -211,3 +210,17 @@ func (mc *MigrationController) syncGameServer(ctx context.Context, key string) e
 
 	return nil
 }
+
+func (mc *MigrationController) anyAddressMatch(node *k8sv1.Node, gs *agonesv1.GameServer) bool {
+	nodeAddresses := []string{}
+	for _, a := range node.Status.Addresses {
+		if a.Address == gs.Status.Address {
+			return true
+		}
+		nodeAddresses = append(nodeAddresses, a.Address)
+	}
+	msg := fmt.Sprintf("GameServer/Node address mismatch: gs.Name=%s: gs.Status.Address=%s, node.Status.Addresses=%s",
+		gs.Name, gs.Status.Address, strings.Join(nodeAddresses, ","))
+	mc.loggerForGameServer(gs).Warn(msg)
```

Comment:
One minor nit: With `logrus`, we tend to use fields for all of the data pieces, e.g.:

```suggestion
	mc.loggerForGameServer(gs).WithField("gs", gs.Name).WithField("gs.Status.Address", gs.Status.Address).WithField("node.Status.Addresses", strings.Join(nodeAddresses, ",")).Warn("GameServer/Node address mismatch")
```

(note that `gs` is already a common field name, so I used that instead of calling the field `gs.Name`, but the rest I copied from your log message).